dffml/source/dataframe.py

```diff
@@ -0,0 +1,71 @@
+"""
+Expose Pandas DataFrame as DFFML Source
+"""
+from collections import OrderedDict
+from typing import Dict, List, AsyncIterator
+
+
+from ..record import Record
+from ..base import config, field
+from ..util.entrypoint import entrypoint
+from .source import BaseSourceContext, BaseSource
+
+
+class DataFrameSourceContext(BaseSourceContext):
+    async def update(self, record: Record):
+
+        df = self.parent.config.dataframe
+
+        # Store feature data
+        feature_columns = self.parent.config.feature_cols
```

Comment:
The feature columns are the columns of the data frame, without the prediction columns from the config structure.